### PR TITLE
ashwalkers now have an incentive to live longer

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
ashwalkers get stronger after 30, 60, and 90 minutes
speed, armor, and fire breath.

## How This Contributes To The Skyrat Roleplay Experience
ashwalkers need to strive to live longer. For story purposes, older ashies should be raising and help teach the newly arisen ashies. This increase in power can create some form of "superiority" over the younger ashies.

## Changelog

:cl:
add: ashies get stronger after 30, 60, and 90 minutes being alive (individually)
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/ashwalkers/code/species/elder.dm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+	human_parent = parent
+	addtimer(CALLBACK(src, .proc/elder_speed), 30 MINUTES)
+	addtimer(CALLBACK(src, .proc/elder_armor), 60 MINUTES)
+	addtimer(CALLBACK(src, .proc/elder_breath), 90 MINUTES)
+
+/datum/component/elder_ash/proc/elder_speed()
+	human_parent.add_movespeed_modifier(/datum/movespeed_modifier/elder_speed)
+	to_chat(human_parent, span_warning("As time passes, you feel that your legs have become stronger-- your walking is more efficient."))
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/ashwalkers/code/species/elder.dm:25]:
```suggestion
	to_chat(human_parent, span_warning("As time passes, you feel that your legs have become stronger... You can now move around more efficiently."))
```

File: modular_skyrat/modules/ashwalkers/code/species/elder.dm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+/datum/component/elder_ash/proc/elder_armor()
+	human_parent.dna.species.armor = 30
+	to_chat(human_parent, span_warning("As time passes, you feel that your scales have hardened-- you can take hits better."))
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/ashwalkers/code/species/elder.dm:29]:
```suggestion
	to_chat(human_parent, span_warning("As time passes, you feel that your scales have hardened... Getting hit now hurts ever so slightly less."))
```

File: modular_skyrat/modules/ashwalkers/code/species/elder.dm [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	addtimer(CALLBACK(src, .proc/elder_speed), 30 MINUTES)
+	addtimer(CALLBACK(src, .proc/elder_armor), 60 MINUTES)
+	addtimer(CALLBACK(src, .proc/elder_breath), 90 MINUTES)
+
+/datum/component/elder_ash/proc/elder_speed()
+	human_parent.add_movespeed_modifier(/datum/movespeed_modifier/elder_speed)
+	to_chat(human_parent, span_warning("As time passes, you feel that your legs have become stronger-- your walking is more efficient."))
+
+/datum/component/elder_ash/proc/elder_armor()
+	human_parent.dna.species.armor = 30
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/ashwalkers/code/species/elder.dm:28]: I feel like a 30% damage reduction has some chances of leading to cases where they can stack armor and get to 100% armor.

Also, is that just on every armor stat, or is that just for brute/burn, or...?
```suggestion
	human_parent.dna.species.armor = 20
```

File: modular_skyrat/modules/ashwalkers/code/species/elder.dm [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	to_chat(human_parent, span_warning("As time passes, you feel that your scales have hardened-- you can take hits better."))
+
+/datum/component/elder_ash/proc/elder_breath()
+	var/datum/action/cooldown/mob_cooldown/fire_breath/granted_action
+	granted_action = new()
+	granted_action.Grant(human_parent)
+	to_chat(human_parent, span_warning("As time passes, an ancient power within you has awakened-- the power of fire is yours!"))
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/ashwalkers/code/species/elder.dm:35]: I'm really not sure about this one.

I think a ritual could be a lot cooler than just "boom you've got firebreath now that you spent the last hour and a half on the same tile as someone else, exchanging emotes".

```suggestion
	to_chat(human_parent, span_warning("As time passes, an ancient power within you has awakened... The power of fire is yours!"))
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Overall, I think the idea to promote them staying alive longer is a good one.

However, here's a counter-proposal to make it more engaging (and arguably cooler too!):

How about going all-in with the shamanism that goes around Ash walkers and making it so each of these upgrades require some sort of ritual (that wouldn't necessarily need to be overly complicated, but that would require another person to do it with you and, like, maybe some herbs and like one regular item from fauna, or whatever) that would take, let's say, five or ten seconds to accomplish, with some potentially cool visual effects going alongside it? I'm sure we can find someone to help you with the visuals, we've got a bunch of very talented spriters that have come up recently.

I know it's a bit more work, but it'd feel more rewarding and more interesting for everyone, even neutral third parties just watching it happen. Basically, it'd shift it more from something you're entitled to get after an arbitrary amount of time, to something that you get rewarded with for doing, even if it's just a simple ceremony (potentially super roleplay-friendly too!)

**Comment by jjpark-kb** [comment]
I do see the solid changes requested, they're great-- I just don't have any more motivation.

**Pull request closed without merging**